lisa/wlgen/rta.py

```diff
@@ -1318,9 +1323,53 @@ class ConcretePropertyBase(PropertyBase):
     barrier and the value will be set again, even if it means repeating the
     same value as earlier.
     """
+    REQUIRED_KCONFIG_KEYS = []
+    """
+    List of KCONFIG keys that need to be =Y on the target kernel for this
+    property to be usable.
+    """
+
+
+    @classmethod
+    def check_kconfig(cls, plat_info, strict=True):
+        """
+        Check whether ``plat_info`` contains the kernel KCONFIG keys contained
+        in :attr:`~ConcretePropertyBase.REQUIRED_KCONFIG_KEYS`.
+
+        :param keys: Kernel config keys to check, e.g. ['CONFIG_FOO_BAR'].
+        :type keys: list(str)
+
+        :param strict: If True, raise an exception if any key is missing.
+                       If False, log a warning if any key is missing.
+        :type strict: bool
+        """
+        def raise_err(msg, exc):
+            if strict:
+                raise exc
+            else:
+                cls.get_logger().warning(msg)
+            return False
+
+        for key in cls.REQUIRED_KCONFIG_KEYS:
+            try:
+                val = plat_info['kernel']['config'][key]
+            except KeyError as e:
+                return raise_err(f'Kernel config does not have key: {key}', e)
+
+            # Dependency could be built as a module, in which case we'd
+            # need to check whether it is loaded. Assert against Y for
+            # now.
+            if val is not KernelConfigTristate.YES:
+                msg = f'Kernel config {key}={val}, expected Y'
+                return raise_err(msg, TargetStableError(msg))
+            return True
```

Comment:
`return True` seems to have one too many indentation level. It will currently be executed at the first iteration of the loop, bypassing the check for the subsequent keys